DisjointGridSet.java

```diff
@@ -0,0 +1,57 @@
+package assignment4;
+
+import java.util.HashSet;
+import java.util.Set;
+
+
+public class DisjointGridSet {
+
+    private int[] parents;
+    private int initialColumnCount;
+

```

Comment:
You don't test here that your grid is rectangular. Always worth adding assertions to your code to check your preconditions haven't broken.